### PR TITLE
Moving pointer logic from klass to pointer.py

### PR DESCRIPTION
**Pull request by cgoxo**
## Description
#5403 
Moving logic of pointer from `klass.py` to `pointer.py`

## Affected Dependencies
`src/syft/core/pointer/pointer.py
`
## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: src/syft/core/pointer/pointer.py [hunk 1]
```diff
@@ -587,7 +594,7 @@ def __del__(self) -> None:
         if self.gc_enabled:
             self.client.gc.apply(self)
 
-    def _resolve_pointer_type(self):
+    def _resolve_pointer_type(self: AbstractPointer) -> AbstractPointer:
```
Comment (cgoxo) [inline src/syft/core/pointer/pointer.py:597]: Hello @tudorcebere!
What should I type instead of `(self: AbstractPointer)` to avoid mypy error?

**Comment by cgoxo** [comment]
Hi @tudorcebere !
I was trying to remove this method from `src/syft/ast/klass.py` but it shows error in `create_pointer_class` under           
`class Class(Callable)`. I tried to call with `Pointer()._resolve_pointer_type` but that did not work.

**Comment by cgoxo** [comment]
@tudorcebere please mention what other methods do I have to shift?

**Comment by cgoxo** [comment]
@tudorcebere Please review it!

**Comment by cgoxo** [comment]
Hello @tudorcebere !
If there is any other method to shift let me know

**Comment by tudorcebere** [comment]
Hi @cgoxo!

We got other plans in terms of refactoring klass.py. Thank you for your help, but we are not 100% on how to proceed, but when we'll do it, we'll add you as a contributor. 👍🏼 

Thank you for your interest, ping me on slack if you want help on picking up new issues (holidays are over) 💯

**Comment by cgoxo** [comment]
Sure, no worries.

**Pull request closed without merging**